app/models/api/v1/report.rb

```diff
@@ -70,18 +71,31 @@ def student_json(student)
     {
       id: student.id,
       name: student.name,
+      first_name: student.first_name,
+      last_name: student.last_name,
       started_offering:  started_offering
     }
   end
 
-  # Helpers that provide student answers grouped by embeddable key:
 
+  def self.update_report_learners(report_learners)
+    report_learners.each do |learner|
+      if learner.last_report.nil? or learner.last_report < MAX_REPORT_LEARNER_AGE.ago
+        learner.last_report = Time.now
+        learner.update_fields()
+      end
+    end
+  end
+
+  # Helpers that provide student answers grouped by embeddable key:
   def get_student_answers
     # Collect all the student answers for given offering.
     student_ids = @students.map { |s| s.id }
-    answers = Report::Learner.where(offering_id: @offering.id, student_id: student_ids).map do |report_learner|
+    report_learners = Report::Learner.where(offering_id: @offering.id, student_id: student_ids)
+    API::V1::Report.update_report_learners(report_learners)
```

Comment:
Hmm. It looks like the report_learner should be initialized as soon as it is created here: https://github.com/concord-consortium/rigse/blob/report-feedback-work-part-2/app/models/portal/learner.rb#L80
That after_create will typically happen when a student clicks on a link to run an activity. That click goes to `offerings#show` which then calls `setup_portal_student`, which then creates the `Portal::Learner`.

So it seems like there should be no partially initialized report_learners.